rasa/utils/plotting.py

```diff
@@ -117,103 +117,134 @@ def plot_confusion_matrix(
         fig.savefig(output_file, bbox_inches="tight")
 
 
-def plot_histogram(
-    hist_data: List[List[float]], title: Text, output_file: Optional[Text] = None
-) -> None:
-    """Plots a side-by-side comparative histogram of the confidence distribution.
+def _paired_histogram_specification(
+    histogram_data: List[List[float]],
+    num_bins: int,
+    density: bool,
+    x_pad_fraction: float,
+    y_pad_fraction: float,
+) -> Tuple[List[float], List[List[float]], List[float], Tuple[float, float]]:
+    """Extracts all information from the data needed to plot a paired histogram.
 
     Args:
-        hist_data: histogram data
-        title: title of the plot
-        output_file: output file to save the plot to
+        histogram_data: Two data vectors
+        num_bins: Number of bins to be used for the histogram
+        density: If true, generate information for a probability density histogram
+        x_pad_fraction: Percentage of extra space in the horizontal direction
+        y_pad_fraction: Percentage of extra space in the vertical direction
+
+    Returns:
+        The bins, values, ranges of either x-axis, and the range of the y-axis
     """
-    import matplotlib.pyplot as plt
-
-    plt.gcf().clear()
+    min_data_value = np.min(np.concatenate(histogram_data))
+    max_data_value = np.max(np.concatenate(histogram_data))
+    bin_width = (max_data_value - min_data_value) / num_bins
+    bins = [min_data_value + i * bin_width for i in range(num_bins + 2)]
+    histograms = [
+        np.histogram(data, bins=bins, density=density)[0] for data in histogram_data
+    ]
+
+    y_padding = 0.5 * bin_width + y_pad_fraction * bin_width
+
+    if density:
+        # When we plot the PDF, let both x-axes run to the same value
+        # so it's easier to compare visually
+        v = max([(1.0 + x_pad_fraction) * max(histogram) for histogram in histograms])
+        x_ranges = [v, v]
+    else:
+        x_ranges = [(1.0 + x_pad_fraction) * max(histogram) for histogram in histograms]
```

Comment:
Could you add a comment here describing how you are selecting the ranges. It's hard to guess without knowing the structure of `histograms`?